### PR TITLE
Adding batchnorm fix for GFX1030

### PR DESCRIPTION
**Pull request by muralinr**
Adding workaround fix for GFX1030 ASM kernel

### CODE REVIEW

File: src/ocl/batchnormocl.cpp [hunk 1]
```diff
@@ -1051,7 +1051,8 @@ void BatchNormBackward(Handle& handle,
                 std::string parms;
 
                 if((n > 64) && (n % 2 == 0) && (variant == 3) && (bfpmixparm) && (useSaved) &&
-                   ctx.use_asm_kernels && ctx.rmv.IsV2orV3())
+                   ctx.use_asm_kernels && ctx.rmv.IsV2orV3() &&
+                   (handle.GetDeviceName() != "gfx1030"))
```
Comment (atamazov) [inline src/ocl/batchnormocl.cpp:1055]:
```suggestion
                   !StartsWith(name, "gfx1"))
```
or
```suggestion
                   (StartsWith(name, "gfx8") || StartsWith(name, "gfx9")))
```

And add `#include <miopen/stringutils.hpp>` at the beginning of file.

**Review by atamazov — CHANGES_REQUESTED** [review]
Let's foresee other gfx10 derivatives and, possibly, other new gfx architectires, at once.

File: src/ocl/batchnormocl.cpp (hunk 1 above)
Comment (muralinr) [reply src/ocl/batchnormocl.cpp]: Ok. Done..

**Review by atamazov — APPROVED** [review]

**Comment by daniellowell** [comment]
@muralinr was this tested on Navi21 and confirmed to solve the issue on the JIRA?

**Comment by muralinr** [comment]
[AMD Official Use Only - Internal Distribution Only]

Yes Daniel.. I tested it on Navi21..

Thanks,
Murali

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Pull request merged**